python/lsst/daf/butler/core/_butlerUri.py

```diff
@@ -1196,14 +1225,61 @@ def as_local(self) -> Tuple[str, bool]:
         return tmpFile.name, True
 
     def read(self, size: int = -1) -> bytes:
-        # Docstring inherits
+        log.debug("Reading from remote resource: %s", self.geturl())
         stream = True if size > 0 else False
-        r = requests.get(self.geturl(), stream=stream)
+        r = self.session.get(self.geturl(), stream=stream)
         if not stream:
             return r.content
         else:
             return next(r.iter_content(chunk_size=size))
 
+    def write(self, data: bytes, overwrite: bool = True) -> None:
+        log.debug("Writing to remote resource: %s", self.geturl())
+        if not overwrite:
+            if self.exists():
+                raise FileExistsError(f"Remote resource {self} exists and overwrite has been disabled")
+        self.session.put(self.geturl(), data=data)
+
+    def transfer_from(self, src: ButlerURI, transfer: str = "copy",
+                      transaction: Optional[Union[DatastoreTransaction, NoTransaction]] = None) -> None:
+        """Transfer the current resource to a Webdav repository.
+
+        Parameters
+        ----------
+        src : `ButlerURI`
+            Source URI.
+        transfer : `str`
+            Mode to use for transferring the resource. Supports the following
+            options: copy.
+        transaction : `DatastoreTransaction`, optional
+            Currently unused.
+        """
+        # Fail early to prevent delays if remote resources are requested
+        if transfer not in self.transferModes:
+            raise ValueError(f"Transfer mode '{transfer}' not supported by URI scheme {self.scheme}")
+
+        log.debug(f"Transferring {src} [exists: {src.exists()}] -> "
+                  f"{self} [exists: {self.exists()}] (transfer={transfer})")
+
+        if self.exists():
+            raise FileExistsError(f"Destination path '{self}' already exists.")
+
+        if transfer == "auto":
+            transfer = self.transferDefault
+
+        if isinstance(src, type(self)):
+            if transfer == "move":
+                self.session.request('MOVE', src.geturl(), headers={'Destination': self.geturl()})
```

Comment:
daf_butler uses double quotes for strings so please fix all the quoting for consistency.